File: core/src/main/java/top/fksoft/server/http/utils/ClassUtils.java
package top.fksoft.server.http.utils;

public class ClassUtils {

    public static String getClassName(int fun) {
        return getElement(fun).getClassName();
    }

    private static StackTraceElement getElement(int fun) {
        StackTraceElement[] stacktrace = Thread.currentThread().getStackTrace();
         return stacktrace[2 + fun];
    }


    public static String getMethodName(int fun) {
        return getElement(fun).getMethodName();
    }
    public static String getFileName(int fun) {
        return getElement(fun).getFileName();
    }
    public static int getLineNumber(int fun) {
        return getElement(fun).getLineNumber();
    }

}

File: core/src/main/java/top/fksoft/server/http/utils/ClassUtils.kt
package top.fksoft.server.http.utils

object ClassUtils {

    fun getClassName(`fun`: Int): String {
        return getElement(`fun`).className
    }

    private fun getElement(`fun`: Int): StackTraceElement {
        val stacktrace = Thread.currentThread().stackTrace
        return stacktrace[2 + `fun`]
    }


    fun getMethodName(`fun`: Int): String {
        return getElement(`fun`).methodName
    }

    fun getFileName(`fun`: Int): String? {
        return getElement(`fun`).fileName
    }

    fun getLineNumber(`fun`: Int): Int {
        return getElement(`fun`).lineNumber
    }

}
